include/lbann/io/persist.hpp

```diff
@@ -145,6 +206,101 @@ bool read_double (int fd, const char *name, double *val);
 bool write_string(int fd, const char *name, const char *buf, size_t size);
 bool read_string(int fd, const char *name, char *buf, size_t size);
 
+class NonexistentArchiveFile : public std::runtime_error {
+public:
+  NonexistentArchiveFile(std::string const& filename) : std::runtime_error(std::string("Archive file not found: ") + filename) {}
+};
+
+template <typename C>
+void write_cereal_archive(C& obj, persist& p, persist_type pt, const std::string& suffix) {
+  std::ofstream os(p.get_filename(pt) + suffix);
+  if(!os.is_open()) {
+    throw NonexistentArchiveFile("Unable to open file for saving checkpoint " + p.get_filename(pt) + suffix);
```

Comment:
1. The argument naming scheme for `NonexistentArchiveFile` clearly indicates that it expects a filename, not an error message.
2. The open could fail for reasons beyond the file not existing. Permission errors, interrupts, etc...
3. Don't we _want_ the file to not exist? Or do we want this to overwrite an existing archive?

Reply:
Right now we want to be able to gracefully handle an execution context not being in a checkpoint and not deserialized.